dynamodb/runtime/pom.xml

```diff
@@ -49,6 +49,25 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>dynamodb-enhanced</artifactId>
```

Comment:
Not sure if we should add it here or if it should be a separate artifact. Depends on if people will massively move to it I suppose or if they will want to use the low level client without bringing this one?

Also, you know I was doing to ask for that :) : we need a test, especially one that also runs in native as we need to make sure it works with native (and native might require some additional work at the extension level).

@scrocquesel do you have an opinion on this?

Reply:
> Not sure if we should add it here or if it should be a separate artifact. Depends on if people will massively move to it I suppose or if they will want to use the low level client without bringing this one?
> 
> Also, you know I was doing to ask for that :) : we need a test, especially one that also runs in native as we need to make sure it works with native (and native might require some additional work at the extension level).
> 
> @scrocquesel do you have an opinion on this?

TBH, i'm not doing dev with aws, so I do not have strong opinion about having this deps in the extension. But as you said, if added, native mode should be thorougly tested and community should be prepared to give enough support if this deps often change. Maybe as a separate extension ?

Reply:
> I added a test for async and sync. Unfortunately it is not working, I always get `software.amazon.awssdk.core.exception.SdkClientException: Unable to instantiate request handler chain for client. Listed request handler ('software.amazon.awssdk.enhanced.dynamodb.internal.ApplyUserAgentInterceptor') does not implement the interface software.amazon.awssdk.core.interceptor.ExecutionInterceptor API.` Maybe the Enhanced Client simply does not work with execution interceptor.
> 
> Any advice is highly appreciated

FYI - If I do not set `quarkus.class-loading.reloadable-artifacts=software.amazon.awssdk:dynamodb-enhanced` I get the NoClassDefFoundError which is stated in the aws sdk issue. 

> Maybe as a separate extension?

If I understood it correctly the dynamodb enhanced client depends on the dynamodb client. https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/dynamodb-enhanced/pom.xml#L130  I have no experience with Quarkus extensions, but if they depend on each other doesn't it make more sense to do this in one extension?